delocate/tools.py

```diff
@@ -229,7 +416,7 @@ def _line0_says_object(line0, filename):
     )
 
 
-def get_install_names(filename):
+def get_install_names(filename: str) -> Tuple[str, ...]:
```

Comment:
Is it tested somewhere, the behavior with multiple architectures?  Sorry if I missed it.

Reply:
It needs to be tested with the version of macOS that returns multiple architectures from its otool.  I couldn't figure out if GitHub Actions has a runner which has that version of the tool.  So I've been relying on feedback from the people at #78.

It'd be nice to know what version of the tools changed their output and what versions of macOS I can expect these version to be on.  Otherwise the best I can do is update the workflow to run the latest available versions of macOS.

Reply:
How about using mock on `subprocess` to simulate the output for multiple architectures?

Reply:
That could work, but the functions which normalize the output of otool are already tested with the alternative output.  I'd rather test on the real tools but I can't even start with that without knowing when the change in output happened.

Reply:
That would be good - to test on the real tools - but I worry that we have a currently untested change of behavior for these two public functions - even if their subfunctions are tested.  I mean, we could change these two public functions, and give a different output, and the tests would not detect this.

Reply:
There isn't a regression or change to any existing functions.  That would show up on the current tests, and I haven't changed any tests to expect a different output.  This is the way I see it:

| | Old behavior| After PR
|--|---|-------
| Old otool output | Works | Works 
| New otool output | Broken according to feedback | Works according to feedback

I actually can't change their behavior and these functions will have to be deprecated and replaced because their current spec is incompatible with how architectures are handled by the new tools.  When new behavior is required to handle a library then `NotImplemendError` will be raised.  So far I found this to be rare, but I expect an issue to be raised eventually.  The corrected version of these functions would look like `_get_install_ids` where they return values for each architecture as a dictionary.

Reply:
Sure - no question this PR is better than the previous situation.  But - it's still true that we currently have changes here, that work for the new otool output, that we could change, and we would not detect this in the tests.  I'm happy to merge as-is, and put in some tests soonish, if you like.

Reply:
>  But - it's still true that we currently have changes here, that work for the new otool output, that we could change, and we would not detect this in the tests. 

I understand that, but there isn't much I can do since I don't know enough about macOS to install the updated tools on the GitHub Runners.

I've done my best in this situation.  I have the changes in output handled by two functions that can test whatever input I want to give them: `_parse_otool_listing` and `_check_ignore_archs`.

`_parse_otool_listing` interprets the output and can handle the old and new output.  With the new output it attaches the architecture to each value since that's now part of otools output.  With the old output it attaches a blank `""` to the value.  The values are the same 'list of lines' as before, but now with an optional architecture as the keys of a dictionary.

`_check_ignore_archs` wraps `_parse_otool_listing` and ensures that the old and new output give the same result.  A single item / single architecture / blank architecture has the single value returned as-is.  With multiple items the values are returned as one only if they're all equivalent, which mimics the expected result of the old tools.  If the values don't match then the calling function needs to handle the output of `_parse_otool_listing` directly so `NotImplementedError` gets raised in this case.

With this the difference in otools output can be handled and tested by these two functions using fake otool output based on feedback and the existing older output.  I have high confidence that this works how I've explained it.

Reply:
Maybe a tl;dr:

`_parse_otool_listing` fixes #78 by parsing both the old and new output.

`_check_ignore_archs` is an adapter for the existing functions.  It returns what the old output would've returned and is tested with both the old and new output.

This means the output is normalized by the time a caller like `get_install_names` gets to handle it.

Reply:
Right - but we're going round in circles a bit here - I'm sure you agree that it's currently possible to change `get_install_names` and `get_rpaths` so that the multi-arch stuff breaks, and the tests continue to pass.  The question is - would you like me to merge now and we'll do some tests later?

Reply:
> The question is - would you like me to merge now and we'll do some tests later?

Sorry.  Yes I'd like it merged its current state.  I can make a new issue just to track that the new tools aren't directly tested yet.  Maybe someone else will bring in with more information.

> I'm sure you agree that it's currently possible to change get_install_names and get_rpaths so that the multi-arch stuff breaks, and the tests continue to pass.

Sorry again, but maybe I'm confused by the phrasing.  These functions are already like this: broken when they'd have to return different results per-architecture, but I haven't heard anyone run into this yet whenever I ask them to install this PR.